scripts/create_django_issue.py

```diff
@@ -0,0 +1,290 @@
+    current_minor_version = DjVersion.parse(current_version_str)
+    newer_versions: set[DjVersion] = set()
+    for django_version in get_django_versions():
+        if django_version > current_minor_version:
+            newer_versions.add(django_version)
+
+    return current_minor_version, sorted(newer_versions, reverse=True)
+
+
+_TABLE_HEADER = """
+
+## {file}.txt
+
+| Name | Version in Master | {dj_version} Compatible Version | OK |
+| ---- | :---------------: | :-----------------------------: | :-: |
+"""
+VITAL_BUT_UNKNOWN = [
+    "django-environ",  # not updated often
+    "pylint-django",  # classifier not included in setup.py
+]
+
+
+class GitHubManager:
+    def __init__(self, base_dj_version: DjVersion, needed_dj_versions: list[DjVersion]):
+        self.github = Github(GITHUB_TOKEN)
+        self.repo = self.github.get_repo(GITHUB_REPO)
+
+        self.base_dj_version = base_dj_version
+        self.needed_dj_versions = needed_dj_versions
+        # (major+minor) Version and description
+        self.existing_issues: dict[DjVersion, Issue] = {}
+
+        # Load all requirements from our requirements files and preload their
+        # package information like a cache:
+        self.requirements_files = ["base", "local", "production"]
+        # Format:
+        # requirement file name: {package name: (master_version, package_info)}
+        self.requirements: dict[str, dict[str, tuple[str, dict]]] = {
+            x: {} for x in self.requirements_files
+        }
+
+    def setup(self) -> None:
+        self.load_requirements()
+        self.load_existing_issues()
+
+    def load_requirements(self):
+        print("Reading requirements")
+        for requirements_file in self.requirements_files:
+            with (REQUIREMENTS_DIR / f"{requirements_file}.txt").open() as f:
+                for line in f.readlines():
+                    if "==" in line and not line.startswith("{%"):
+                        name, version = get_name_and_version(line)
+                        self.requirements[requirements_file][name] = (
+                            version,
+                            get_package_info(name),
+                        )
+
+    def load_existing_issues(self):
+        """Closes the issue if the base Django version is greater than needed"""
+        print("Load existing issues from GitHub")
+        qualifiers = {
+            "repo": GITHUB_REPO,
+            "author": "app/github-actions",
+            "state": "open",
+            "is": "issue",
+            "in": "title",
+        }
+        issues = list(
+            self.github.search_issues(
+                "[Django Update]", "created", "desc", **qualifiers
+            )
+        )
+        print(f"Found {len(issues)} issues matching search")
+        for issue in issues:
+            matches = re.match(r"\[Update Django] Django (\d+.\d+)$", issue.title)
+            if not matches:
+                continue
```

Comment:
It may perhaps help if we also check `issue.author` being `dependabot[bot]` or something similar. This ensures we don't duplicate issues or try to incorrectly edit a description with the assumption that the bot creates unique issue titles.

Reply:
I should have mentioned that the search linked in my comment isn't exactly like the one we're making in the code. The one from the code should only return results for issues opened by GitHub actions thanks to `"author": "app/github-actions"`. 

I think that if the title matches (on top of the other query params), we're should be quite sure to get the correct one. Am I missing another another use case?

Reply:
Oh I see, I didn't read your comment properly (I assumed you were saying the current version of the regex would produce false positive). It LGTM!